overlord/state/change_test.go

```diff
@@ -929,3 +929,120 @@ func (ts *taskRunnerSuite) TestAbortUnreadyLanes(c *C) {
 		c.Assert(strings.Join(obtained, " "), Equals, strings.Join(expected, " "), Commentf("setup: %s", test.setup))
 	}
 }
+
+// setup is a list of tasks "<task1> <task2>", order is <task1>-><task2>
+// (implies task2 waits for task 1)
+var cyclicDependencyTests = []struct {
+	setup string
+	order string
+	err   string
+}{
+
+	// Some basics.
+	{
+		setup: "t1",
+	}, {
+		setup: "",
+	}, {
+		// independent tasks
+		setup: "t1 t2 t3",
```

Comment:
Can we have a variant of this test with 1 independent tasks and a few tasks with dependencies?

Reply:
added